java/vector/src/main/java/org/apache/arrow/vector/complex/NonNullableStructVector.java

```diff
@@ -79,6 +81,40 @@ public NonNullableStructVector(String name, BufferAllocator allocator, FieldType
     this.valueCount = 0;
   }
 
+  @Override
+  public void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers) {
+    if (ownBuffers.size() != 0) {
+      throw new IllegalArgumentException("Illegal buffer count, expected " + 0 + ", got: " + ownBuffers.size());
```

Comment:
not sure this add much without a constant variable?

Reply:
The changes in this class were moved from the subclass, for when I was trying to use this class as a FieldVector for the map data. It didn't work out, so I don't think I can revert these but it is actually the same in all vectors, which I agree could be cleaned up.